### PR TITLE
Get rid of environment-latest and update to python 3.9

### PR DESCRIPTION
**Pull request by jsignell**
This is an alternative to #8259. The failures in upstream tests match those on main.

Closes https://github.com/dask/dask/issues/6808

### CODE REVIEW

**Comment by raybellwaves** [comment]
> @raybellwaves I think you added environment-latest, so just wanted to get your thoughts on this.

Go for it.

File: dask/tests/test_utils.py
Comment (jsignell) [reply dask/tests/test_utils.py:771]: yeah, I couldn't decide where to put it. That is a good suggestion.

File: continuous_integration/environment-3.8-dev.yaml
Comment (jsignell) [reply continuous_integration/environment-3.8-dev.yaml:1]: yeah good point! I uncommented bcolz in environment.yaml. That's all I saw.

File: continuous_integration/environment-3.9.yaml
Comment (jrbourbeau) [inline continuous_integration/environment-3.9.yaml:33]: This change is causing conda solve issues in CI since `bcolz` still isn't available on Python 3.9 yet (and I don't know if it ever will be). Since `bcolz` is [no longer maintained](https://github.com/Blosc/bcolz#unmaintained-package-notice), we might consider deprecating support for it in Dask.

File: dask/tests/test_docs.py
Comment (jrbourbeau) [inline dask/tests/test_docs.py:9]: We might also want to add `.github/workflows/upstream.yml` to ensure our `upstream` CI build is running against the latest support Python version too

**Review by jrbourbeau — COMMENTED** [review]
Thanks for putting this together @jsignell. It looks like there's general interest in this change. I've left a couple of small final comments, but otherwise this looks good

File: continuous_integration/environment-3.9.yaml
Comment (jsignell) [reply continuous_integration/environment-3.9.yaml:33]: oh strange. I could have sworn I checked that it was available.

Comment (jrbourbeau) [reply continuous_integration/environment-3.9.yaml:33]: No worries -- FWIW I opened up https://github.com/dask/dask/issues/8330 to see if there's interest in deprecating `from_bcolz`

**Review by jrbourbeau — APPROVED** [review]
Thanks @jsignell. The CI failures here are unrelated and being addressed in https://github.com/dask/dask/pull/8336. After that is in, I'll merge `main` here, then merge this PR once CI finishes

**Pull request merged**
